src/Aspire.Dashboard/Otlp/WebApplicationExtensions.cs

```diff
@@ -0,0 +1,85 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using Aspire.Dashboard.Authentication;
+using Aspire.Dashboard.Otlp.Grpc;
+using Microsoft.AspNetCore.Mvc;
+using OpenTelemetry.Proto.Collector.Logs.V1;
+using OpenTelemetry.Proto.Collector.Metrics.V1;
+using OpenTelemetry.Proto.Collector.Trace.V1;
+
+namespace Aspire.Dashboard.Otlp;
+
+public static class WebApplicationExtensions
+{
+    // https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/large-object-heap
+    private const int MaxSizeLessThanLOH = 84999;
+
+    public static WebApplication ConfigureHttpOtlp(this WebApplication app, Uri httpEndpoint)
+    {
+        app.MapPost("/v1/metrics",
+                async ([FromServices] OtlpMetricsService service, HttpContext httpContext) =>
+                {
+                    return await ExportOtlpData(httpContext,
+                            sequence => service.Export(ExportMetricsServiceRequest.Parser.ParseFrom(sequence), null!))
+                        .ConfigureAwait(false);
```

Comment:
Is this API accepting a protobuf request message and then returning a JSON response message? I couldn't see any Protobuf serialization code so I assume that the MapPost func is returning a value and ASP.NET Core is doing the serialization, which is always JSON.

I need to read the specification for OTLP+HTTP. I know it supports both Protobuf and JSON, but both in one request seems very weird. I think to support both the API method would have to look at the request content-type and do some content negotiation. I've yet to do that sort of thing inside minimal API methods so I don't know how hard/easy it is.

Also, I'm surprised the Protobuf response message just works when serialized to JSON if the .NET serializer is just accepting a plain old C# class and serializing it. Protobuf has rules about how it maps Protobuf messages to JSON (and unfortunately OTLP+HTTP/JSON breaks some of them). Maybe it works because the export response message is so simple.

Reply:
I think it will be much easier to initially only support OTLP+HTTP with Protobuf. Supporting JSON will require a lot of custom JSON serialization.

Reply:
The OTLP .NET SDK  supports only **HTTP/Protobuf** and **grpc** (https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md). I suggest to add HTTP/JSON support when OLTP .NET does also support it.

Reply:
Regarding the return value, you a probably right. I have to check the spec here.
The OTLP .NET HTTP Exporter pretty much just checks the status code of the response and stores the headers in bad case:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/043d8a31c768d05544c06629e072492238a0069b/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/BaseOtlpHttpExportClient.cs#L43-L57